src/mapeo-project.js

```diff
@@ -501,6 +504,63 @@ export class MapeoProject {
   get $icons() {
     return this.#iconApi
   }
+
+  /** @params {Object} opts
+   * @property {String} path
+   */
+  async importConfig({ configPath }) {
+    // console.log(path.resolve(configPath))
+    try {
+      await fs.stat(path.resolve(configPath))
+    } catch (e) {
+      console.log(`error loading config file ${configPath}`, e)
+    }
+    const zip = await yauzl.open(configPath)
+    const ids = await this.preset.getMany()
+    if (ids.length !== 0) await this.preset.delete(ids)
+    try {
+      for await (const entry of zip) {
+        if (entry.filename === 'presets.json') {
+          /* eslint-disable no-unused-vars */
+          const { fields, presets } = await readPresets(
+            await zip.openReadStream(entry)
+          )
+          for (let [fieldName, field] of Object.entries(fields)) {
+            const fieldDoc = {
+              // shouldn't schemaName be derived when calling .create?
+              schemaName: 'field',
```

Comment:
Here I'm adding `schemaName` manually, but it should be redundant. In `dataType[kCreateWithDocId]` we can get `schemaName` from `this.#schemaName` and add it to the `value` directly

Reply:
The advantage of keeping it is that ensures stricter typing - another schema could have the same types for other properties, which could leave an implementer calling with wrong object. It's a way of making explicit which type we are using. I'm open to arguments for implying it and removing it from the `create` type though.